packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js

```diff
@@ -305,29 +337,48 @@ export default compose(
 			isEmptyDefaultBlock,
 			isFirst: firstIndex === 0,
 			isLast: lastIndex === blockOrder.length - 1,
+			isReusableBlockType,
+			reusableBlock,
 			rootClientId,
 			selectedBlockClientId,
 			selectedBlockPossibleTransformations,
 		};
 	} ),
 	withDispatch(
-		( dispatch, { clientIds, rootClientId, currentIndex }, { select } ) => {
+		(
+			dispatch,
+			{ clientIds, rootClientId, currentIndex, selectedBlockClientId },
+			{ select }
+		) => {
 			const {
 				moveBlocksDown,
 				moveBlocksUp,
 				duplicateBlocks,
 				removeBlocks,
 				insertBlock,
 				replaceBlocks,
+				clearSelectedBlock,
 			} = dispatch( blockEditorStore );
 			const { openGeneralSidebar } = dispatch( 'core/edit-post' );
 			const { getBlockSelectionEnd, getBlock } = select(
 				blockEditorStore
 			);
 			const { createSuccessNotice } = dispatch( noticesStore );
 
+			const {
+				__experimentalConvertBlockToStatic: convertBlockToStatic,
+			} = dispatch( reusableBlocksStore );
+
 			return {
 				createSuccessNotice,
+				convertToRegularBlocks() {
+					clearSelectedBlock();
+					// Convert action is executed at the end of the current JavaScript execution block
+					// to prevent issues related to undo/redo actions.
+					setImmediate( () =>
```

Comment:
I tested these changes in a test branch (`rnmobile/try/reusable-block-feature`) where it also includes the changes of [Update Reusable block to use the inner block PR](https://github.com/WordPress/gutenberg/pull/30966) and I spotted that on Android the following flow wasn't working:
1. Convert a reusable block to regular blocks
2. Undo the changes
3. Convert again the same reusable block to regular blocks
4. The conversion is not being done ❌ 

For this reason, I decided to add the following changes:
1. Unselect the block to prevent potential issues when the reusable block is replaced by the regular blocks.
2. Execute the convert action at the end of the current JS execution block by using `setImmediate`. I tried to investigate the root cause of this issue but I didn't manage to find it. As far as I checked after the convert action is executed, calling the selector `getBlock` with the reusable block's client id always returns `null` 🤷‍♂️ .